tests/compile/piecewise/test_toy_llama.py

```diff
@@ -66,10 +72,12 @@ def __init__(self, config: LlamaConfig) -> None:
             bias=False,
         )
 
-        self.gate_up_projection.weight.data.fill_(0.0)
-        self.down_projection.weight.data.fill_(0.0)
+        nn.init.eye_(self.gate_up_projection.weight.data[:config.mlp_size])
+        nn.init.eye_(self.gate_up_projection.weight.data[config.mlp_size:])
+        nn.init.eye_(self.down_projection.weight.data)
```

Comment:
Why don't we use random initialization? I know it might cause some numerical issues, but I think we should make the test more rigorous.

Reply:
Or, we maybe we can provide two options? One for tractability and another for testing correctness.

Reply:
using eye init, we can compute the output directly, without needing to compare against two runs (which can be non-deterministic)

Reply:
Yeah so let's provide two options? - eye for debugging, and random for more rigorous correctness test?

I think the correctness guarantee is so important here. Normal people cannot debug this at all.

Reply:
added both tractable tests and random seed tests.